src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Microsoft.NET.Sdk.Razor.CodeGeneration.targets

```diff
@@ -70,10 +70,15 @@ Copyright (c) .NET Foundation. All rights reserved.
     </ItemGroup>
   </Target>
 
+  <Target
+    Name="_RazorEnsureCompiled"
+    Condition="!Exists('$(_RazorGenerateTargetReferenceAssembly)')"
+    DependsOnTargets="Compile" />
+
   <Target
     Name="ResolveTagHelperRazorGenerateInputs"
-    DependsOnTargets="_EnsureRazorTasksAssemblyDefined;Compile"
-    Inputs="$(MSBuildAllProjects);@(RazorReferencePath)"
+    Inputs="$(MSBuildAllProjects);@(RazorReferencePath);$(_RazorGenerateTargetReferenceAssembly)"
```

Comment:
Assuming the changes to places where `@(RazorReferencePath)` are due to removing the 'app' assembly from it so you can do things differently depending on phase. LMK if that is not correct.

Reply:
Yes, this is correct. When doing code generation, we use either `obj\Debug\netcoreapp3.0\MyAssembly.dll` or `obj\Debug\netcoreapp3.0\RazorDefinition\MyAssembly.dll` (the temp assembly) depending on where it's being called. During `RazorCompile` we always use `obj\Debug\netcoreapp3.0\MyAssembly.dll` since it's the "correct" assembly.